src/pipx/venv_metadata_inspector.py

```diff
@@ -12,88 +40,114 @@
+
+    return new_package
+
+
+def get_package_dependencies(
+    dist: metadata.Distribution, extras: Set[Any]
+) -> List[Requirement]:
+    # Add an empty extra to enable evaluation of non-extra markers
+    if not extras:
+        extras.add("")
+    dependencies = []
+    for req in map(Requirement, dist.requires or []):
+        if not req.marker:
+            dependencies.append(req)
+        else:
+            for extra in extras:
+                if req.marker.evaluate({"extra": extra}):
+                    dependencies.append(req)
+                    break
+
+    return dependencies
+
+
+def get_apps(dist: metadata.Distribution, bin_path: Path) -> List[str]:
     apps = set()
-    for section in ["console_scripts", "gui_scripts"]:
-        # "entry_points" entry in setup.py are found here
-        for name in pkg_resources.get_entry_map(dist).get(section, []):
-            if (bin_path / name).exists():
-                apps.add(name)
-            if WINDOWS and (bin_path / (name + ".exe")).exists():
-                # WINDOWS adds .exe to entry_point name
-                apps.add(name + ".exe")
-
-    if dist.has_metadata("RECORD"):
-        # for non-editable package installs, RECORD is list of installed files
-        # "scripts" entry in setup.py is found here (test w/ awscli)
-        for line in dist.get_metadata_lines("RECORD"):
-            entry = line.split(",")[0]  # noqa: T484
-            path = (Path(dist.location) / entry).resolve()
-            try:
-                if path.parent.samefile(bin_path):
-                    apps.add(Path(entry).name)
-            except FileNotFoundError:
-                pass
-
-    if dist.has_metadata("installed-files.txt"):
-        # not sure what is found here
-        for line in dist.get_metadata_lines("installed-files.txt"):
-            entry = line.split(",")[0]  # noqa: T484
-            path = (Path(dist.egg_info) / entry).resolve()  # type: ignore
-            try:
-                if path.parent.samefile(bin_path):
-                    apps.add(Path(entry).name)
-            except FileNotFoundError:
-                pass
+
+    sections = {"console_scripts", "gui_scripts"}
+    # "entry_points" entry in setup.py are found here
+    for ep in dist.entry_points:
+        if ep.group not in sections:
+            continue
+        if (bin_path / ep.name).exists():
+            apps.add(ep.name)
+        if WINDOWS and (bin_path / (ep.name + ".exe")).exists():
+            # WINDOWS adds .exe to entry_point name
+            apps.add(ep.name + ".exe")
+
+    # search installed files
+    # "scripts" entry in setup.py is found here (test w/ awscli)
+    for path in dist.files or []:
+        dist_file_path = Path(dist.locate_file(path)).resolve()
+        try:
+            if dist_file_path.parent.samefile(bin_path):
+                apps.add(path.name)
+        except FileNotFoundError:
+            pass
+
+    # not sure what is found here
```

Comment:
Is this still the case? I wonder if we can remove this section.

Reply:
I wondered too...but I was also really worried about corner cases because the universe of strange python packages is vast.

I wonder if this duplicates the above for loop `for path in dist.files`.  Maybe one of the other python packaging experts can give some guidance?

Reply:
@uranusjr , @gaborbernat , do either of you know if `installed-files.txt` is only used for Python 2.*?  It's hard to find documentation on it, but what I find all seems to refer to Python 2.*.  If `installed-files.txt` is only used for Python 2.* then we can definitely remove it.

Reply:
I don’t think it has anything to do with Python 2/3. AFAIK `installed-files.txt` is a setuptools/distutils artifact from the old days before `.dist-info` was specified. Newer setuptools versions should never write this.

Reply:
I find references to ancient pip generating this file (there is a mention in changelog for pip 0.3 https://pip.pypa.io/en/stable/news/#id556)

Reply:
Regardless of the library of origin, it seems very old.  If we are updating our shared libraries this code that views `installed-files.txt` seems like it can be removed.

Let me know if there are any objections, otherwise I will remove it.

Reply:
I didn't read the code closely; can someone explain what it actually does? While new(-ish) packaging tools would never generate the file, it may still be present in systems based on Python packaging but never bothered to update.

Reply:
It just goes through every line in file `installed-files.txt` and sees if any of them is in the venv `bin/` subdirectory.  Basically exactly the same thing as the for loop above it does by searching the iterable `Distribution.files`.

If this is part of ancient `pip`, I'm wondering if it was created each time on install on the client machine, and not made as part of the package distribution?

Reply:
One of the few mentions I can find on the internet, from a setuptools Issue discussing a possible solution (June 2015):
https://github.com/pypa/setuptools/issues/371#issuecomment-202918638
> Just have setuptools add installed-files.txt to the egg-info directory. pip has been doing this for a long time before .dist-info was on the radar.